src/history/history.rs

```diff
@@ -244,13 +245,27 @@ impl History {
 
         like_query.push_str("%");
 
-        let query = "SELECT id, cmd, cmd_tpl, session_id, when_run, exit_code, selected, dir, rank,
+        let query = "WITH matched_commands AS (
+                           SELECT id, cmd, cmd_tpl, session_id, when_run, exit_code, selected, dir, rank,
```

Comment:
Does using this `when_run` not work, such that you need to do the second query?  If this isn't the most recent, maybe  when the contextual_commands temp table is made, it should be `MAX(when_run)`?

Reply:
There would need to be at least two queries here to get the max grouped only by `cmd` but on looking temp table creation is a _much_ better place for this!